File: utbot-go/src/main/kotlin/org/utbot/go/gocodeanalyzer/AnalysisTargets.kt
package org.utbot.go.gocodeanalyzer

internal data class AnalysisTarget(
    val absoluteFilePath: String,
    val targetFunctionNames: List<String>,
    val targetMethodNames: List<String>
)

internal data class AnalysisTargets(val targets: List<AnalysisTarget>)

File: utbot-go/src/main/resources/go_source_code_analyzer/analysis_targets.go
package main

type AnalysisTarget struct {
	AbsoluteFilePath    string   `json:"absoluteFilePath"`
	TargetFunctionNames []string `json:"targetFunctionNames"`
	TargetMethodNames   []string `json:"targetMethodNames"`
}

type AnalysisTargets struct {
	Targets []AnalysisTarget `json:"targets"`
}
